File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s)
    {
        string word="";
        vector<string> out;
        for(int i=0;i<s.length();i++)
        {
            if(s[i]!=' ')
            {
                word+=s[i];
            }
            else
            {
                if(!word.empty())
                {
                    out.push_back(word);
                    word="";
                }
            }
        }
        if(!word.empty())
        {
            out.push_back(word);
        }
        string result;
        for(int i=out.size()-1;i>=0;i--)
        {
            result+=out[i];
            if(i>0)
            {
                result+=" ";
            }
        }
        return result;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        word=""
        wordlist=[]
        for i in range(len(s)):
            if s[i]!=' ':
                word+=s[i]
            else:
                if word!="":
                    wordlist.append(word)
                    word=""
        if(word!=""):
            wordlist.append(word)
        
        result=' '.join(reversed(wordlist))
        return result